tasks/GenerateReadme.cfc

```diff
@@ -0,0 +1,202 @@
+/**
+* Generate README.md files for exercises
+*/
+component  {
+
+	/**
+	* Run the task to generate read me files
+	*/
+	function run(
+		slug,
+		boolean all=false,
+		boolean overwrite=false
+	){
+
+		/* Init global variables */
+		repoRootPath = expandPath( getDirectoryFromPath( getCurrentTemplatePath() ) & '../' );
+		problemSpecsPath = expandPath( getDirectoryFromPath( getCurrentTemplatePath() ) & '../../problem-specifications' );
+		trackReadmeTemplate = fileRead( repoRootPath & "config/exercise_readme.go.tmpl" );
+		trackReadmeInsert= fileRead( repoRootPath & "docs/EXERCISE_README_INSERT.md" );
+		overwriteExistingFiles = arguments.overwrite;	
+
+		if( !directoryExists( problemSpecsPath ) ){
+			error( 'Cannot find [problem-specifications] repo.  It needs to be git cloned to [#problemSpecsPath#]' );
+		}
+		
+		if( all ) {
+			
+			// Get an array of all the excercise names
+			var exercises = directoryList( expandPath( getDirectoryFromPath( getCurrentTemplatePath() ) & '../exercises' ) )
+				.map( function( path ) {
+					return path.listLast( '/\' );
+				} );
+		
+		} else {
+		
+			// Acquire slug
+			if( isNull( arguments.slug ) ) {
+				arguments.slug = ask( 'Exercise slug as defined by Excercism. Ex: "hello-world": ' );
+			}
+			var exercises = [ slug ];
+			
+		}
+
+		exercises.each( function( slug ) {
+			
+			try {
+				generateReadme( arguments.slug );
+
+			} catch( 'commandException' var e ) {
+				print.boldRedLine( e.message )
+					.line();
+			}
+			
+		} ); // End exercises.each()
+	
+	}
+
+	private function generateReadme( required string slug ) {
+
+		var readmeContent = "";
+		var readMeFile = repoRootPath & "/exercises/" & arguments.slug & "/README.md";
+
+		if ( !fileExists(readMeFile) || overwriteExistingFiles) {
+
+			var spec = getSpec( arguments.slug );
+			readmeContent = replace( trackReadmeTemplate, "{{ .Spec.Name }}" , spec.name );
+			readmeContent = replace( readmeContent, "{{ .Spec.Description -}}" , spec.description );
+			readmeContent = reReplace( readmeContent, "{{\- with \.Hints }}\n{{ \. }}\n{{ end }}" , getHints( arguments.slug ) );
+			readmeContent = reReplace( readmeContent, "{{\- with \.TrackInsert }}\n{{ \. }}\n{{ end }}" , trackReadmeInsert );
+			if ( len( spec.credits ) ) {
+				readmeContent = reReplace( readmeContent, "{{\- with \.Spec\.Credits \-}}\n(#### Source\n\n){{ \. }}\n{{ end }}" , "\1" & spec.credits & chr( 10 ) );
+			}
+			else {
+				readmeContent = reReplace( readmeContent, "{{\- with \.Spec\.Credits \-}}\n#### Source\n\n{{ \. }}\n{{ end }}" , "" );
+			}
+			/* Write the file */
+			fileWrite( readMeFile , readmeContent );
+
+			print.greenLine( 'Readme generated!' )
+				.indentedYellowLine( readMeFile )
+				.toConsole();
+		}
+		else {
+			print.redLine("README.md already exists! Use the overwrite=true to overwrite the file.")

```

Comment:
Technically, the syntax when calling the task runner would be:
```
:overwrite=true
```
or 
```
--:overwrite
```
Maybe just say, "use the `overwrite` flag" if not using actual copy-pastable syntax.  Also note, I've fixed some bugs in Task Runner args on the bleeding edge of ComandBox so `overwrite=true` might work right now, but it's only on accident and will be "fixed" in version 3.9.0 of CommandBox :)

Also, for what it's worth, I usually call flags like this `force` just for consistency.  You don't have to change it, but just something to keep in mind since `force` is idiomatic of pretty much every other CommandBox command that has the option to overwrite something.

Reply:
I think it will be better to keep it consistent. I will change it to force.